TicketsDemo.Mongo/Repositories/MongoTrainRepository.cs

```diff
@@ -0,0 +1,62 @@
+﻿using MongoDB.Driver;
+using System.Collections.Generic;
+using System.Linq;
+using MongoDB.Bson;
+using TicketsDemo.Data.Entities;
+using TicketsDemo.Data.Repositories;
+
+namespace TicketsDemo.Mongo
+{
+    public class MongoTrainRepository : ITrainRepository
+    {
+        private Train TrainInitialization(Train train)
```

Comment:
private functions to be defined 'below' public ones in order to make class interface easily easier to track.